### PR TITLE
Further outgoing message improvements

### PR DESCRIPTION
**Pull request by rowanseymour**
* Added partner field to outgoing model which we'll need for any partner based view of sent messages
* Reworked case timeline so it doesn't rely on fetching messages from backend

### CODE REVIEW

File: casepro/backend/rapidpro.py [hunk 1]
```diff
@@ -316,26 +316,26 @@ def unflag_messages(self, org, messages):
...
-                'flow': msg.type == 'flow',
-                'sender': None,
-                'broadcast': msg.broadcast
+                'direction': Outgoing.DIRECTION,
+                'case': None,
+                'sender': None
             }
 
-        return [remote_as_json(m) for m in remote_messages]
+        return [remote_as_json(m) for m in remote_messages if m.direction == 'out']
```
Comment (rowanseymour) [inline casepro/backend/rapidpro.py]: For CasePro/RapidPro, we'll now rely on local messages entirely for incoming and only use this fetch to pick up additional sent messages from RapidPro

File: casepro/cases/tests.py [hunk 2]
```diff
@@ -633,7 +621,7 @@ def test_timeline(self, mock_push_outgoing, mock_fetch_contact_messages):
         self.assertEqual(len(response.json['results']), 0)
 
         # messages won't have been fetched from the backend this time
-        mock_fetch_contact_messages.assert_not_called()
```
Comment (rowanseymour) [inline casepro/cases/tests.py]: This doesn't exist in Python 2.7

File: casepro/contacts/models.py [hunk 3]
```diff
@@ -241,7 +241,7 @@ def release(self):
         self.is_active = False
         self.save(update_fields=('is_active',))
 
-    def as_json(self, full=False):
+    def as_json(self, full=True):
```
Comment (rowanseymour) [inline casepro/contacts/models.py]: Convention for all these `as_json(..)` methods is that by default they return complete representation - and `full=False` returns the shortened representation typically used when object is embedded in another object.

File: casepro/msgs/models.py [hunk 4]
```diff
@@ -452,17 +452,16 @@ def create(cls, org, user, activity, text, contacts, urns, case=None):
 
     def as_json(self):
```
Comment (rowanseymour) [inline casepro/msgs/models.py]: Was previously trying to make JSON representation of `Outgoing` match that of `Message` but these models continue to diverge so it's not worth it.

**Comment by ericnewcomer** [comment]
weee!

**Pull request merged**